R/enron_process_data.R

```diff
@@ -1,4 +1,120 @@
+#' Processes Enron dataset to long format.
+#'
+#' @param file The file path of the \code{.xls} file for processing.
+#'
+#' @return Returns a \code{data.frame}.
 #' @export
-enron_process_data = function(file) {
+#'
+#' @examples
+#' enron_process_data(enron_download_file())
+enron_process_data <- function(file) {
+  names_df <-
+    readxl::read_xls(file, range = "B3:C31") |>
+    dplyr::rename(type = 1, location = 2) |>
+    tidyr::fill(type)
 
+
+  full_df <- readxl::read_xls(file, col_names = FALSE, skip = 0)
+
+  binary_df <- !is.na(full_df)
+
+  left_most_col_num <- 6
+  right_most_col_num <- ncol(full_df)
+
+  vec <-
+    binary_df |>
+    as.data.frame.matrix() |>
+    dplyr::mutate(dplyr::across(dplyr::everything(), as.numeric)) |>
+    dplyr::pull(left_most_col_num) # first column of the grids of values starting from row 35
+
+
+  sequence <- c(0, 1, 1, 0)
+
+  vec_match <-
+    zoo::rollapply(
+      vec,
+      FUN = function(vector, sequence) {
+        all(vector == sequence)
```

Comment:
This searches through the xls file to look for the monthly tables. The sequence it looks for is the header of the tables so it will continue to find more tables (if more months are added) rather than only the ones that are currently included. Could definitely do with some more comments here for clarity though.